pkg/provider/aws/knowledgebase/cloudfront.go

```diff
@@ -0,0 +1,112 @@
+package knowledgebase
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/core"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base"
+	"github.com/klothoplatform/klotho/pkg/multierr"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
+)
+
+var CloudfrontKB = knowledgebase.Build(
+	knowledgebase.EdgeBuilder[*resources.CloudfrontDistribution, *resources.S3Bucket]{
+		Expand: func(distro *resources.CloudfrontDistribution, bucket *resources.S3Bucket, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			var errs multierr.Error
+			for consRef := range distro.ConstructsRef {
+				conn := s3ToCloudfrontConnection{
+					distro:    distro,
+					bucket:    bucket,
+					dag:       dag,
+					construct: consRef,
+				}
+				oai, err := conn.createOai()
+				if err != nil {
+					errs.Append(err)
+					continue
+				}
+				err = conn.attachPolicy(oai)
+				errs.Append(err)
+			}
+			return errs.ErrOrNil()
+		},
+		Configure: func(distro *resources.CloudfrontDistribution, bucket *resources.S3Bucket, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			distro.DefaultRootObject = bucket.IndexDocument
+			return nil
+		},
+	},
+	knowledgebase.EdgeBuilder[*resources.CloudfrontDistribution, *resources.OriginAccessIdentity]{},
+)
+
+type s3ToCloudfrontConnection struct {
+	distro    *resources.CloudfrontDistribution
+	bucket    *resources.S3Bucket
+	dag       *core.ResourceGraph
+	construct core.AnnotationKey
+}
+
+func (conn s3ToCloudfrontConnection) createOai() (*resources.OriginAccessIdentity, error) {
+	oai, err := core.CreateResource[*resources.OriginAccessIdentity](conn.dag, resources.OriginAccessIdentityCreateParams{
+		Name: fmt.Sprintf("%s-%s", conn.bucket.Name, conn.construct.ID),
+		Refs: core.AnnotationKeySetOf(conn.construct),
+	})
+	if err != nil {
+		return nil, err
+	}
+	conn.dag.AddDependency(conn.distro, oai)
+
+	// This should be in an edge Configure, but it requires all three of the AOI, bucket, and distro -- so it's easier
+	// to do it here, at create time when we already have all three.
+	s3OriginConfig := resources.S3OriginConfig{
+		OriginAccessIdentity: core.IaCValue{
+			Resource: oai,
+			Property: resources.CLOUDFRONT_ACCESS_IDENTITY_PATH_IAC_VALUE,
+		},
+	}
+	origin := &resources.CloudfrontOrigin{
+		S3OriginConfig: s3OriginConfig,
+		DomainName: core.IaCValue{
+			Resource: conn.bucket,
+			Property: resources.BUCKET_REGIONAL_DOMAIN_NAME_IAC_VALUE,
+		},
+		OriginId: conn.construct.ID,
+	}
```

Comment:
just a helper method right? If so i like that, it will really only ever be used in one place but thats ok

Reply:
Yup, exactly. It's mainly to keep the code clean and self-documenting, since it's doing a few things.